zebra-consensus/src/transaction.rs

```diff
@@ -154,6 +197,42 @@ impl Request {
     }
 }
 
+impl Response {
+    /// The verified mempool transaction, if this is a mempool response.
+    pub fn into_mempool_transaction(self) -> Option<VerifiedUnminedTx> {
+        match self {
+            Response::Block { .. } => None,
+            Response::Mempool { transaction, .. } => Some(transaction),
+        }
+    }
```

Comment:
Maybe a `From` impl ?

```suggestion

impl From<Response> for Option<VerifiedUnminedTx> {
    fn from(self) -> Self {
        match self {
            Response::Block { .. } => None,
            Response::Mempool { transaction, .. } => Some(transaction),
        }
    }
}

impl Response {
```

Reply:
I'd like this conversion to be explicit, so we don't accidentally convert a mempool response in the block verifier.

(And if we do, the reviewer can see that it's happening.)